lib/containers/common.pm

```diff
@@ -49,9 +63,21 @@ sub install_docker_when_needed {
     assert_script_run('docker info');
 }
 
-sub clean_docker_host {
-    assert_script_run('docker stop $(docker ps -q)', 180) if script_output('docker ps -q | wc -l') != '0';
-    assert_script_run('docker system prune -a -f',   180);
+sub allow_registry_suse_de_for_docker {
+    # Allow our internal 'insecure' registry
+    assert_script_run("mkdir -p /etc/docker");
+    assert_script_run('cat /etc/docker/daemon.json; true');
+    assert_script_run(
+        'echo "{ \"insecure-registries\" : [\"registry.suse.de\", \"registry.suse.de:443\", \"registry.suse.de:5000\"] }" > /etc/docker/daemon.json');
+    assert_script_run('cat /etc/docker/daemon.json');
+    systemctl('restart docker');
+}
+
+sub clean_container_host {
+    my $runtime = $_[0];
+    die "You must define the runtime!" unless $runtime;
```

Comment:
Other methods don't check their arguments for validity either

Reply:
yeah, good point. Maybe we can add a die as well in a later PR as improvement.